WalletWasabi/WabiSabi/Client/AliceClient.cs

```diff
@@ -15,238 +15,236 @@ namespace WalletWasabi.WabiSabi.Client;
-
-	public Money EffectiveValue => SmartCoin.EffectiveValue(FeeRate, IsPayingZeroCoordinationFee ? CoordinationFeeRate.Zero : CoordinationFeeRate);
+    private AliceClient(
+        Guid aliceId,
+        RoundState roundState,
+        ArenaClient arenaClient,
+        SmartCoin coin,
+        OwnershipProof ownershipProof,
+        IEnumerable<Credential> issuedAmountCredentials,
+        IEnumerable<Credential> issuedVsizeCredentials,
+        bool isPayingZeroCoordinationFee)
+    {
+        var roundParameters = roundState.CoinjoinState.Parameters;
+        AliceId = aliceId;
+        RoundId = roundState.Id;
+        ArenaClient = arenaClient;
+        SmartCoin = coin;
+        OwnershipProof = ownershipProof;
+        FeeRate = roundParameters.MiningFeeRate;
+        CoordinationFeeRate = roundParameters.CoordinationFeeRate;
+        IssuedAmountCredentials = issuedAmountCredentials;
+        IssuedVsizeCredentials = issuedVsizeCredentials;
+        MaxVsizeAllocationPerAlice = roundParameters.MaxVsizeAllocationPerAlice;
+        ConfirmationTimeout = roundParameters.ConnectionConfirmationTimeout / 2;
+        IsPayingZeroCoordinationFee = isPayingZeroCoordinationFee;
+    }
+
+    public Guid AliceId { get; }
+    public uint256 RoundId { get; }
+    private ArenaClient ArenaClient { get; }
+    public SmartCoin SmartCoin { get; }
+    private OwnershipProof OwnershipProof { get; }
+    private FeeRate FeeRate { get; }
+    private CoordinationFeeRate CoordinationFeeRate { get; }
+    public IEnumerable<Credential> IssuedAmountCredentials { get; private set; }
+    public IEnumerable<Credential> IssuedVsizeCredentials { get; private set; }
+    private long MaxVsizeAllocationPerAlice { get; }
+    private TimeSpan ConfirmationTimeout { get; }
+    public bool IsPayingZeroCoordinationFee { get; }
+
+    public static async Task<AliceClient> CreateRegisterAndConfirmInputAsync(
+        RoundState roundState,
+        ArenaClient arenaClient,
+        SmartCoin coin,
+        IKeyChain keyChain,
+        RoundStateUpdater roundStatusUpdater,
+        CancellationToken registrationCancellationToken,
+        CancellationToken confirmationCancellationToken)
+    {
+        AliceClient? aliceClient = null;
+        try
+        {
+            aliceClient = await RegisterInputAsync(roundState, arenaClient, coin, keyChain, registrationCancellationToken).ConfigureAwait(false);
+            await aliceClient.ConfirmConnectionAsync(roundStatusUpdater, confirmationCancellationToken).ConfigureAwait(false);
+
+            Logger.LogInfo($"Round ({aliceClient.RoundId}), Alice ({aliceClient.AliceId}): Connection was confirmed.");
+        }
+        catch (OperationCanceledException)
+        {
+            if (aliceClient is { })
+            {
+                // Unregistering coins is only possible before connection confirmation phase.
+                await aliceClient.TryToUnregisterAlicesAsync(registrationCancellationToken).ConfigureAwait(false);
```

Comment:
I replaced `confirmationCancellationToken` with `registrationCancellationToken`.  

 `registrationCancellationToken` cancelled if
1. asked from the outside
2. wrong phase exception
3. registration times out + expiry time

In contrast `confirmationCancellationToken` times out if
1. asked from the outside
2. wrong phase exception is conn conf
3. conn conf times out + expiry time

I think this is how it's correct.